gluten-core/src/test/scala/org/apache/gluten/MarkdownBuilder.scala

```diff
@@ -0,0 +1,127 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.gluten
+
+import com.vladsch.flexmark.formatter.Formatter
+import com.vladsch.flexmark.parser.{Parser, ParserEmulationProfile, PegdownExtensions}
+import com.vladsch.flexmark.profile.pegdown.PegdownOptionsAdapter
+import com.vladsch.flexmark.util.data.{MutableDataHolder, MutableDataSet}
+import com.vladsch.flexmark.util.sequence.SequenceUtils.EOL
+
+import scala.collection.mutable.ListBuffer
+
+class MarkdownBuilder {
+  private val buffer = new ListBuffer[String]
+
+  /**
+   * append a single line with replacing EOL to empty string
```

Comment:
Nit: 
comment style, capitalize the initial letter and end up with period mark. Please also check other comments.